package-lock.json

```diff
@@ -37829,6 +37971,7 @@
 			"version": "7.1.2",
 			"resolved": "https://registry.npmjs.org/glob/-/glob-7.1.2.tgz",
 			"integrity": "sha512-MJTUg1kjuLeQCJ+ccE4Vpa6kKVXkPYJ2mOCQyUuKLcLQsdrMCpBPUi8qVE6+YuaJkozeA9NusTAw3hLr8Xe5EQ==",
+			"dev": true,
```

Comment:
Hmm, what do we want to do with the `"dev": true` diffs? There are both removals and additions, but I can't think of an efficient way to tell if any are real changes (in the sense of a dep that shouldn't be loaded in production mode). Thoughts?

Reply:
@hypest at a high level, I am not too concerned about these changes as I do not believe there to be large risk of significant damage anywhere. That said, I looked into this a little bit more. 

![GitHub's package-lock.json diff presentation can be misleading](https://user-images.githubusercontent.com/438664/145888385-e541e2e4-3f72-4b95-a686-302468e8f0e1.png)

First, GitHub's diff of the `package-lock.json` can be misleading and difficult to parse correctly. For example, the above image might be easily interpreted as `fbjs-css-vars` received a new `"dev": true` flag. However, from reviewing the raw file itself, that is not accurate. 

Second, it would appear a small number of packages had their `dev` flag modified. 

### Added
- `path-parse`
- `resolve`

### Removed
- `@babel/helper-remap-async-to-generator`
- `@babel/helper-wrap-function`
- `@babel/plugin-transform-async-to-generator`
- `parse-url`
- `slash`

Of those, I looked into the `@babel` changes. It would appear `metro-react-native-babel-preset`, which _is not_ a `"dev": true` dependency, [added `@babel/plugin-transform-async-to-generator` in `v0.65.1`](https://github.com/facebook/metro/commit/eb06fbdcfba9ed3ce03350d76519359ab98891a9). In this PR, we upgraded from `v0.64.0` to `v0.66.2`, which would explain why each of the `@babel` packages — which are dependent upon one another — removed their [`"dev": true` flag](https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json#dev).

Reply:
I see, thanks for looking into this deeper David! I remember the "dev:true" diffs becoming an annoyance to gutenberg-mobile folks that were not sure what to do with the change and we often attribute it to misalignment of npm versions (against the version used to commit the lock file). I'm not really concerned either about the diff itself but I wonder if you think there might be a npm version misalignment at play here 🤔.

Reply:
> I'm not really concerned either about the diff itself but I wonder if you think there might be a npm version misalignment at play here 🤔.

I definitely won't rule it out in general. The usage of Lerna and `npm@6` in this monorepo repository definitely creates odd outcomes frequently when adding/modify/removing packages. 

That said, given that I was able to find a fairly reasonable explanation for the changes to the `@babel` packages, no, my thought is that these are legitimate lock file changes triggered from changes within dependencies we upgraded. 

Happy to research of the aforementioned list to verify them, though.